tests/ConsoleWriter.cs

```diff
@@ -4,15 +4,28 @@ namespace TemporalioSamples.Tests;
 
 public class ConsoleWriter : StringWriter
 {
-    private ITestOutputHelper output;
+    private readonly ITestOutputHelper output;
+#pragma warning disable CA2213 // We don't want to dispose original output
+    private readonly TextWriter originalOut;
+#pragma warning restore CA2213
 
-    public ConsoleWriter(ITestOutputHelper output)
+    public ConsoleWriter(ITestOutputHelper output, TextWriter originalOut)
     {
         this.output = output;
+        this.originalOut = originalOut;
     }
 
     public override void WriteLine(string? value)
     {
-        output.WriteLine(value);
+        try
+        {
+            output.WriteLine(value);
+        }
+        catch (Exception ex) when (ex is InvalidOperationException && ex.Message.Contains("no currently active test"))
+        {
+            // Fall back to original console output when no test is active or writer is disposed
+            // This happens when background tasks write to console
+            originalOut.WriteLine(value);
```

Comment:
Unclear why this is necessary. It's complaining on one of the console log lines in `RunDemoAsync`, which by definition finishes before the test does. I think there's some weird thread context switching going on or something. Open to better ways to fix this if you have any.

Reply:
Hrmm, looking at https://github.com/xunit/xunit/issues/2377 and other links referencing that, can you try updating all 3 xunit deps in the tests csproj to latest (or at least 2.9.0) and see if this still occurs?